activity/mqtt/activity.go

```diff
@@ -0,0 +1,83 @@
+package mqtt
+
+import (
+	"github.com/eclipse/paho.mqtt.golang"
+	"github.com/project-flogo/core/activity"
+	"github.com/project-flogo/core/data/coerce"
+	"github.com/project-flogo/core/data/metadata"
+)
+
+var MqttActivitymd = activity.ToMetadata(&Settings{}, &Input{}, &Output{})
+
+func init() {
+	_ = activity.Register(&MqttActivity{}, New)
+}
+
+func New(ctx activity.InitContext) (activity.Activity, error) {
+	settings := &Settings{}
+	err := metadata.MapToStruct(ctx.Settings(), settings, true)
+	if err != nil {
+		return nil, err
+	}
+
+	options := initClientOption(settings)
+	if err != nil {
+		//ctx.Logger().Errorf("Kafka parameters initialization got error: [%s]", err.Error())
+		return nil, err
+	}
+
+	mqttClient := mqtt.NewClient(options)
+
+	if token := mqttClient.Connect(); token.Wait() && token.Error() != nil {
+		return nil, token.Error()
+	}
+
+	act := &MqttActivity{settings: settings, client: mqttClient}
+	return act, nil
+}
+
+type MqttActivity struct {
+	settings *Settings
```

Comment:
why do you need settings here?

Reply:
After moving Topic and Qos to settings, I'll need it here. I do not see any way to access the client's topic or qos.